server/test/tuist/runs_test.exs

```diff
@@ -844,4 +844,243 @@ defmodule Tuist.RunsTest do
       assert url == nil
     end
   end
+
+  describe "cas_output_metrics/1" do
+    test "returns correct metrics for build with CAS outputs" do
+      # Given
+      {:ok, build} =
+        RunsFixtures.build_fixture(
+          cas_outputs: [
+            %{
+              node_id: "node1",
+              checksum: "abc123",
+              size: 2000,
+              duration: 2000,
+              compressed_size: 1600,
+              operation: :download
+            },
+            %{
+              node_id: "node2",
+              checksum: "def456",
+              size: 2000,
+              duration: 2000,
+              compressed_size: 1600,
+              operation: :download
+            },
+            %{
+              node_id: "node3",
+              checksum: "ghi789",
+              size: 2000,
+              duration: 2000,
+              compressed_size: 1600,
+              operation: :upload
+            }
+          ]
+        )
+
+      # When
+      # Add a small delay to ensure ClickHouse has processed the data
+      Process.sleep(100)
```

Comment:
Don't we have a polling-based utility somewhere else in the codebase? I'd use that one instead.